packages/shared-core/src/filters.ts

```diff
@@ -307,7 +310,133 @@ export class ColumnSplitter {
  * Builds a JSON query from the filter structure generated in the builder
  * @param filter the builder filter structure
  */
-export const buildQuery = (filter: SearchFilter[]) => {
+
+const builderFilter = (expression: SearchFilter) => {
```

Comment:
Could you explain the different usages of these functions now? `builderFilter` suggests it's only used in the builder, and looks to convert the old filter structure into a lucene query. This was used for in memory searching, so does in memory searching work with the new filter structure?

Reply:
`builderFilter`, definitely a typo. This was the core of the old filtering, but its now been repurposed to only process the conditions in the new grouped search request. I'll rename this to `buildCondition`. Nice spot.